Content.Server/Language/LanguageSystem.cs

```diff
@@ -129,20 +129,18 @@ public void RemoveLanguage(
         EntityUid uid,
         string language,
         bool removeSpoken = true,
-        bool removeUnderstood = true,
-        LanguageKnowledgeComponent? knowledge = null,
-        LanguageSpeakerComponent? speaker = null)
```

Comment:
Do not remove those. If it bothers you so much, make them into `Entity<LanguageSpeakerComponent?, LanguageKnowledgeComponent?>`.